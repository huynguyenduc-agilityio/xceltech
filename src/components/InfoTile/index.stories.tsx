import type { Meta, StoryObj } from '@storybook/react';

// Icons
import {
  BagIcon,
  BookIcon,
  DocumentIcon,
  MailIcon,
  MoneyIcon,
  UserGroupIcon,
  UserIcon,
} from '@/icons';

// Components
import InfoTile from '.';

const meta = {
  title: 'Components/InfoTile',
  component: InfoTile,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <div className="flex justify-center w-[1000px] p-9 bg-blue-light">
      <InfoTile {...args} />
    </div>
  ),
} satisfies Meta<typeof InfoTile>;

export default meta;

type Story = StoryObj<typeof InfoTile>;

export const Messages: Story = {
  args: {
    title: 'Messages',
    count: 4,
    icon: <MailIcon width={100} height={68} />,
    styled: 'bg-secondary text-black-default',
  },
};

export const Jobs: Story = {
  args: {
    title: 'Jobs',
    count: 1,
    icon: <BagIcon width={92} height={74} />,
    styled: 'bg-primary',
  },
};

export const Candidates: Story = {
  args: {
    title: 'Candidates',
    count: 3000,
    icon: <UserGroupIcon width={106} height={74} />,
    styled: 'bg-green-primary',
  },
};

export const Resumes: Story = {
  args: {
    title: 'Resumes',
    count: 200000,
    icon: <DocumentIcon width={57} height={75} />,
    styled: 'bg-black-muted',
  },
};

export const Employees: Story = {
  args: {
    title: 'Employees',
    count: 3000000,
    icon: <UserIcon width={83} height={83} />,
    styled: 'bg-secondary text-black-default',
  },
};

export const Leaves: Story = {
  args: {
    title: 'Leaves',
    count: 1234567890,
    icon: <BookIcon width={103} height={62} />,
    styled: 'bg-primary',
  },
};

export const Payrolls: Story = {
  args: {
    title: 'Payrolls',
    count: 7,
    icon: <MoneyIcon width={99} height={67} />,
    styled: 'bg-green-primary',
  },
};
