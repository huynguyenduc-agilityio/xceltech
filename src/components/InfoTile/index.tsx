import { ReactNode } from 'react';

// Utils
import { cn, formatNumberWithMetricPrefix } from '@/utils';

export interface InfoTileProps {
  count: number;
  title: string;
  icon: ReactNode;
  styled?: string;
}

const InfoTile = ({ count, title, icon, styled }: InfoTileProps) => {
  return (
    <div
      className={cn(
        'flex items-center gap-8 w-[323px] text-white rounded-[14px] pt-9 pb-8 px-8',
        styled,
      )}
    >
      {icon}
      <div className="flex flex-col items-center">
        <span className="text-4xl font-bold leading-[100%]">
          {formatNumberWithMetricPrefix(count)}
        </span>
        <span className="text-lg font-bold">{title}</span>
      </div>
    </div>
  );
};

export default InfoTile;
