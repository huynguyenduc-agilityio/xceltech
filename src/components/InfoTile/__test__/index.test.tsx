import { render } from '@testing-library/react';

// Components
import InfoTile from '..';

describe('InfoTile component', () => {
  it('renders the InfoTile with props', () => {
    const { getByText } = render(
      <InfoTile title="Messages" count={4} icon={<span>icon</span>} />,
    );
    expect(getByText('Messages')).toBeInTheDocument();
    expect(getByText('4')).toBeInTheDocument();
    expect(getByText('icon')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <InfoTile title="Messages" count={4} icon={<span>icon</span>} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
