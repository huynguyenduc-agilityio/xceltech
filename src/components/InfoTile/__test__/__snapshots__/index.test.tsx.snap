// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTile component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-8 w-[323px] text-white rounded-[14px] pt-9 pb-8 px-8"
  >
    <span>
      icon
    </span>
    <div
      class="flex flex-col items-center"
    >
      <span
        class="text-4xl font-bold leading-[100%]"
      >
        4
      </span>
      <span
        class="text-lg font-bold"
      >
        Messages
      </span>
    </div>
  </div>
</DocumentFragment>
`;
